sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -5921,7 +5921,7 @@ pi_result _pi_queue::synchronize() {
     return PI_SUCCESS;
 
   // For in-order queue just wait for the last command.
-  if (isInOrderQueue()) {
+  if (isInOrderQueue() && !LastCommandEvent->IsDiscarded) {
```

Comment:
Why? Isn't `ZeEvent` still there and can be synced?

Reply:
Oh, I guess this is because it may not be host-scope right?

Reply:
Yes, this is one of the reasons. And the main reason is that this event may be in reset state. Basically we can't explicitly synchronize discarded pi_event objects.

Reply:
OK, but please add some comment about it.